### PR TITLE
Merge openxla-pjrt-plugin into integrations/pjrt for in-tree development.

### PR DESCRIPTION
**Pull request by stellaraccident**
Per discussion on https://groups.google.com/g/iree-discuss/c/0pCKa3h4dEU

### CODE REVIEW

File: integrations/pjrt/python_packages/iree_cuda_plugin/setup.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+# Copyright 2023 The OpenXLA Authors
```
Comment (benvanik) [inline integrations/pjrt/python_packages/iree_cuda_plugin/setup.py:3]: some of these files are "The OpenXLA Authors" - is that true/something true going forward?

File: integrations/pjrt/src/iree_pjrt/common/compiler.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_PJRT_PLUGIN_PJRT_COMMON_COMPILER_H_
+#define IREE_PJRT_PLUGIN_PJRT_COMMON_COMPILER_H_
+
+#include <memory>
+#include <string>
+
+#include "iree_pjrt/common/debugging.h"
+// TODO: Excise.
```
Comment (benvanik) [inline integrations/pjrt/src/iree_pjrt/common/compiler.h:14]: looks excised?

File: integrations/pjrt/src/iree_pjrt/common/dylib_entry_point.cc.inc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdio.h>
+
+#include <cstring>
+
+#include "iree/base/internal/path.h"  // TODO: Make public trampoline.
+#include "iree_pjrt/common/api_impl.h"
+//#include "iree/vm/api.h"
```
Comment (benvanik) [inline integrations/pjrt/src/iree_pjrt/common/dylib_entry_point.cc.inc:13]: unused?

File: integrations/pjrt/src/iree_pjrt/partitioner_api/loader/loader_test.c [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "iree_pjrt/partitioner_api/embedding_api.h"
+
+// // static int flagCount = 0;
+
+// // static void printFlag(const char *flag, size_t length, void *userData) {
```
Comment (benvanik) [inline integrations/pjrt/src/iree_pjrt/partitioner_api/loader/loader_test.c:18]: is there value in keeping all this dead code?

File: integrations/pjrt/src/iree_pjrt/common/iree_compiler.cc [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    auto* error = ireeCompilerSessionSetFlags(session_, 1, &flag);
+    if (error) {
+      SetError(error);
+      return false;
+    }
+    return true;
+  }
+
+  // TODO: Excise: Cannot dep on an internal XLA structure.
+  // bool SetFlags(xla::CompileOptions options) override {
```
Comment (benvanik) [inline integrations/pjrt/src/iree_pjrt/common/iree_compiler.cc:99]: worth keeping given this isn't called/will change?

**Review by benvanik — APPROVED** [review]

File: integrations/pjrt/python_packages/iree_cuda_plugin/setup.py (hunk 1 above)
Comment (stellaraccident) [reply integrations/pjrt/python_packages/iree_cuda_plugin/setup.py:3]: I've been slapped enough that I don't touch copyrights. Someone from the google side is legally able to change this and can do it in a followup.

@jpienaar

**Comment by stellaraccident** [comment]
I'm going to land with all of the dead code since it represents things that I had to remove to get it compiling and needs to be re-implemented. Easier to keep the breadcrumb for the very immediate future.

**Pull request merged**